lumentum.py

```
@@ -850,6 +889,68 @@ def get_ports_info(self):
+                elif 4101 <= port_id <= 4120:
+                    info["input-power"] = float(get_text(state.get("input-power"), 0))
+                # DEMUX ports (5201–5220): only output power
+                elif 5201 <= port_id <= 5220:
+                    info["output-power"] = float(get_text(state.get("output-power"), 0))
+
+                self.port_info[str(port_id)] = info
+
+            return self.port_info
+
+        except Exception as e:
+            log_error("get_ports_info RPC error: %s", e)
```

Ruff (0.12.2)

`log_error` may be undefined, or defined from star imports

(F405)